doc/ERROR_CONCEPT.md

```diff
@@ -0,0 +1,250 @@
+
+Reasons to keep it:
+* Gives users a quicker overview about the error (eg. permission error/ resource error/ etc. or another type of categorization such as Markus
+suggested with temporary/permanent/conflict/etc.)
+* Can indicate the type of solution a user has to approach
+
+Reasons against it:
+* Extra line
+* What to do with "uncategorizable" errors
+* Maintenance effort: introduce a new category once there are enough similar errors and change all old errors to this category?
+* Awareness of developers: All devs have to know all categories and choose the correct one. Devs should also be prohibited to throw their error
+into the "uncategorized" category due to lazyness.
+
+Suggestion:
+This line is by far the most discussable one. I thought about categorizations such as
+* Permission/ Resource/ Parse/ External(C++, Ruby, Java)/.../ Miscellaneous:
+Most errors are resource errors (could not stat, no connection, could not close/open/find etc.) This again would not give users
+a particularly useful error indication since he/she has to lookup in the reason anyway. Even subdividing the `Resource` category
+would lead to the same problem. No matter how I look at it, the user
+will have to read the concrete reason in most cases which makes the Description line not that valuable any longer.
+* temporary/permanent/conflict/validation:
+This seems to be a better approach because it includes a possible solution for the user. How to categorize certain errors though
+seems impossible because there can be multiple reasons where to categorize an error. Take `Could not open file` as an example:
+if it is on a NFS and it is offline, *temporary* would be correct. If the file has too restrictive permissions, *permanent* would be more
+appropriate since the admin has to fix it. If the file is currently in use and semlocked by the system, *conflict* would be appropriate.
+* core kdb error/plugin error:
+Not much value for the user
+* differentiation to core features of elektra: Specification/ Notification/ KDB Core/ Bindings/ ...
+An error is again difficult to put into one bucket since some errors might be affected by multiple features (could not stat file for example
+which is used by multiple features).
+This approach would lead to the same problem as we have now (many errors, difficult to maintain)
+
+Personally I do not see how the benefit of a categorization outweighs the drawbacks because in by far the most cases users will have to look
+into the reason in more detail anyway. It should be enough for users to deal with the error just by the reason.
+
+As a result I would suggest to remove the description line.
+
+### Line 2: Error code
+
+Reasons to keep it:
+* Probably could be used to categorize errors such it is done in the http protocol
+* Good to use for shellrecorder scripts to test for errors
+* Possibility for automation (eg. if a certain error number occurred you could automatically trigger certain actions)
+
+Reasons against it:
+* Extra line
```

Comment:
It does not need an extra line, it is only a number.

Reply:
Where does this message then come from:
`Sorry, the error (#...) occurred ;(`

Reply:
Yes, the line exists now, but it does not have to be that way.